tex2pdf_service/tex2pdf/remote_call.py

```diff
@@ -0,0 +1,73 @@
+"""Compile a tarball via the service URL to an outcome file."""
+
+import json
+import logging
+import os
+import tarfile
+import time
+
+import requests
+
+
+def get_outcome_meta(outcome_file: str) -> dict:
+    """Open a compressed outcome tar archive and get the metadata."""
+    meta = {}
+    with tarfile.open(outcome_file, "r:gz") as outcome:
+        for name in outcome.getnames():
+            if name.startswith("outcome-") and name.endswith(".json"):
+                meta_contents = outcome.extractfile(name)
+                if meta_contents:
+                    meta.update(json.load(meta_contents))
+    return meta
+
+
+def submit_tarball(service: str, tarball: str, outcome_file: str, tex2pdf_timeout: int, post_timeout: int, auto_detect: bool = False) -> bool:
+    """Submit tarball to compilation service."""
+    if os.path.exists(outcome_file):
+        raise FileExistsError(f"Outcome file {outcome_file} already exists!")
+    os.makedirs(os.path.dirname(outcome_file), exist_ok=True)
+    logging.info("File: %s", os.path.basename(tarball))
+    meta = {}
+    status_code = None
+
+    with open(tarball, "rb") as data_fd:
+        uploading = {"incoming": (os.path.basename(tarball), data_fd, "application/gzip")}
+        while True:
```

Comment:
`while True` may be inviting an infinite loop during the worst of times (some DDoS 504 situation). How about the retry pattern of:
```python
retries = 2
for attempt in range(1+retries):
  # do work
```
or some variation on that.